homeassistant/components/waze_travel_time/__init__.py

```diff
@@ -147,38 +153,90 @@ async def async_get_travel_times(
             real_time=realtime,
             alternatives=3,
         )
+        _LOGGER.debug("Got routes: %s", routes)
+
+        incl_routes: list[CalcRoutesResponse] = []
+
+        def should_include_route(route: CalcRoutesResponse) -> bool:
+            if len(incl_filters) < 1:
+                return True
+            for street_name in route.street_names:
+                for incl_filter in incl_filters:
+                    if incl_filter in {"", street_name}:
+                        return True
+            _LOGGER.debug(
+                "Excluding route [%s], because no inclusive filter matched any streetname",
+                route.name,
+            )
+            return False
+
+        incl_routes = [route for route in routes if should_include_route(route)]
+
+        filtered_routes: list[CalcRoutesResponse] = []
 
-        if incl_filter not in {None, ""}:
-            routes = [
-                r
-                for r in routes
-                if any(
-                    incl_filter.lower() == street_name.lower()  # type: ignore[union-attr]
-                    for street_name in r.street_names
-                )
-            ]
-
-        if excl_filter not in {None, ""}:
-            routes = [
-                r
-                for r in routes
-                if not any(
-                    excl_filter.lower() == street_name.lower()  # type: ignore[union-attr]
-                    for street_name in r.street_names
-                )
-            ]
-
-        if len(routes) < 1:
+        def should_exclude_route(route: CalcRoutesResponse) -> bool:
```

Comment:
If you add as an argument whether to look at incl or excl, you could simplify this into a single method

Reply:
I can't follow. The two filters work differently. Especially for the debug logging telling on exclude which exclude filter filtered out the route

Reply:
Basically both methods should include and should exclude do the same, they just operate on a different list. Add the list as an argument to the method, and a string to say whether your including or excluding in the log message.
The rest of the code is basically route for route in routes if filter_routes(route, incl filters) and not filter_routes(route, excl filters)

Hope this is clear. Typing from my phone